subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/modifiers/ComponentType.java

```diff
@@ -17,16 +17,26 @@
 package org.gradle.buildinit.plugins.internal.modifiers;
 
 public enum ComponentType {
-    BASIC,
-    APPLICATION,
-    LIBRARY,
-    GRADLE_PLUGIN() {
+    BASIC("app"),
+    APPLICATION("app"),
+    LIBRARY("lib"),
+    GRADLE_PLUGIN("plugin") {
         @Override
         public String toString() {
             return "Gradle plugin";
         }
     };
```

Comment:
I think I'm basically OK with these conventions, except...

1. Maybe we should drop "basic" or only generate the root project since we have no idea what kind of project is needed. At the very least, I think we shouldn't call it an "app" when nothing is in it. 
2. I find it a little weird that we use the generic names app and lib for everything. If a user takes the next step and adds another library, I think we would recommend that they name it something meaningful (and not lib2 or another-lib). I don't have a better suggestion for now, so I think we should go with this.